src/rust/ide/lib/view/src/graph_editor.rs

```diff
@@ -999,17 +1027,28 @@ impl GraphEditorModel {
         let logger         = Logger::new("GraphEditor");
         let display_object = display::object::Instance::new(&logger);
         let nodes          = Nodes::new(&logger);
-//        let visualizations = Stage::new(scene.clone_ref(), Logger::new("VisualisationCollection"));
+//      let visualizations = Stage::new(scene.clone_ref(), Logger::new("VisualisationCollection"));
         let edges          = default();
         let frp            = FrpInputs::new(network);
         let touch_state    = TouchState::new(network,&scene.mouse.frp);
-        let project_name   = component::ProjectName::new(scene,focus_manager);
-        display_object.add_child(&project_name);
-        let screen = scene.camera().screen();
-        let margin = 10.0;
-        project_name.set_position(Vector3::new(0.0,screen.height / 2.0 - margin,0.0));
-        let scene = scene.clone_ref();
-        Self {logger,display_object,scene,cursor,nodes,edges,touch_state,frp,project_name}//visualizations }
+        let breadcrumbs    = component::Breadcrumbs::new(scene,focus_manager);
+        let scene          = scene.clone_ref();
+        let network        = network.clone_ref();
+        Self {logger,display_object,scene,cursor,nodes,edges,touch_state,frp,breadcrumbs
+            ,network}.init()//visualizations }
+    }
+
+    fn init(self) -> Self {
+        let network     = &self.network;
+        let screen      = self.scene.camera().screen();
+        let scene       = self.scene.clone_ref();
+        let breadcrumbs = self.breadcrumbs.clone_ref();
+        self.add_child(&self.breadcrumbs);
+        self.breadcrumbs.set_position(Vector3(-screen.width,screen.height,0.0)/2.0);
+        frp::extend! {network
+            eval scene.frp.shape((shape) breadcrumbs.set_position(Vector3(-shape.width,shape.height,0.0)/2.0));
```

Comment:
This should not be the resposnibility of graph editor to keep breadcrumbs in the right position. Breadcrumbs should be a component which manages its position automatically, so it should start watching that during initialization of ITSELF, not during init of graph editor.